src/main/java/org/folio/des/repository/criteria/CQL2JPACriteria.java

```diff
@@ -90,16 +89,27 @@ private CriteriaQuery<E> toCriteria(CQLNode node)
 
   private void processSort(CQLSortNode node) throws CQLFeatureUnsupportedException {
     List<Order> orders = new ArrayList<>();
+
     for (ModifierSet sortIndex : node.getSortIndexes()) {
       final CqlModifiers modifiers = new CqlModifiers(sortIndex);
-      orders.add(
-          CqlSort.DESCENDING.equals(modifiers.getCqlSort())
-              ? builder.desc(root.get(sortIndex.getBase()))
-              : builder.asc(root.get(sortIndex.getBase())));
+      String jsonPath = sortIndex.getBase();
+      int fieldNamesSize = JsonbNodeConverter.getFieldNames(jsonPath).size();
+
+      Expression<String> field = fieldNamesSize > 1 ?
+        jsonbNodeConverter.convertToExpression(root, jsonPath, builder) : root.get(jsonPath);
```

Comment:
Please add comments that it was before, they are very helpful to understand what is going on

Reply:
Ok, I have added meaningful comments to this place, Thanks